x-pack/plugins/enterprise_search/public/applications/app_search/utils/results/index.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Result, ResultMeta } from '../../components/result/types';
+
+export const flattenField = (fieldName: string, fieldValue: object): Array<[string, object]> => {
+  const flattened: Array<[string, object]> = [];
+
+  if (typeof fieldValue === 'object' && !Object.keys(fieldValue).includes('raw')) {
+    Object.entries(fieldValue).map(([propName, value]) => {
+      flattened.push([fieldName + '.' + propName, value]);
+    });
+  } else {
+    flattened.push([fieldName, fieldValue]);
+  }
+
+  return flattened;
+};
+
+export const flattenSearchResult = (result: Result): Result => {
+  const flattened: Result = {
+    id: { raw: '' },
+    _meta: {
+      id: '',
+      engine: '',
+    },
+  };
+
+  Object.entries(result).map(([key, value]) => {
```

Comment:
Will we support more than 2 levels of nesting ?

Reply:
I believe we should. As you can see this is a draft. Still a lot to do.

Reply:
Cool, in that case the approach in general look solid, need a few tweaks and that should work. 👍